packages/apps/crm/src/screens/event/EventFormScreen.js

```diff
@@ -0,0 +1,181 @@
+    const _defaultEndDate = new Date(
+      new Date().setHours(new Date().getHours() + 1),
+    ).toISOString();
+    const _default = {
+      typeSelect: EventType.category.Meeting,
+      statusSelect: EventType.status.Planned,
+      startDateTime: _defaultStartDate,
+      endDateTime: _defaultEndDate,
+      user: user,
+    };
+
+    if (event != null) {
+      return {
+        ..._default,
+        ...event,
+      };
+    } else if (lead != null) {
+      return {
+        ..._default,
+        eventLead: lead,
+        leadReadonly: true,
+        isLead: true,
+        relatedToSelect: MODELS.lead,
+        relatedToSelectId: lead?.id,
+      };
+    } else if (prospect != null) {
+      return {
+        ..._default,
+        partner: prospect,
+        partnerReadonly: true,
+        hideContactPartner: true,
+        isProspect: true,
+        relatedToSelect: MODELS.partner,
+        relatedToSelectId: prospect?.id,
+      };
+    } else if (client != null) {
+      return {
+        ..._default,
+        partner: client,
+        partnerReadonly: true,
+        isPartner: true,
+        relatedToSelect: MODELS.partner,
+        relatedToSelectId: client?.id,
+      };
+    } else if (contact != null) {
+      return {
+        ..._default,
+        contactPartner: contact,
+        partnerReadonly: true,
+        partner: contact?.mainPartner,
+        contactPartnerReadonly: true,
+        isContact: true,
+        relatedToSelect: MODELS.partner,
+        relatedToSelectId: contact?.id,
+      };
+    }
+
+    return _default;
+  }, [user, event, lead, prospect, client, contact]);
+
+  const createEventAPI = useCallback(
+    (_event, dispatch) => {
+      dispatch(createEvent({event: _event}));
+      if (_event?.isLead) {
+        navigation.navigate('LeadDetailsScreen', {
+          idLead: _event?.eventLead?.id,
+        });
+      }
+      if (_event?.isProspect) {
+        navigation.navigate('ProspectDetailsScreen', {
+          idProspect: _event?.partner?.id,
+        });
+      }
+      if (_event?.isPartner) {
+        navigation.navigate('ClientDetailsScreen', {
+          idClient: _event?.partner?.id,
+        });
+      }
+      if (_event?.isContact) {
+        navigation.navigate('ContactDetailsScreen', {
+          idContact: _event?.contactPartner?.id,
+        });
+      }
+    },
+    [navigation],
+  );
+
+  const updateEventAPI = useCallback(
+    (_event, dispatch) => {
+      let dataToSend = {
+        ..._event,
+      };
+      if (_event?.eventLead != null) {
+        dataToSend = {
+          ..._event,
+          relatedToSelect: MODELS.lead,
+          relatedToSelectId: _event?.eventLead?.id,
+        };
+      } else if (_event?.contactPartner != null || _event?.partner != null) {
+        dataToSend = {
+          ..._event,
+          relatedToSelect: MODELS.partner,
+          relatedToSelectId: _event?.partner?.id,
+        };
+      }
```

Comment:
I think this should be done in the form view with the dependsOn attribute, it will be cleaner as you will only have to send event data without re-working it